src/components/inputModal.tsx

```diff
@@ -1,27 +1,162 @@
-import { Modal, Button } from "react-bootstrap";
-import React from "react";
-import { useState } from "react";
+import { Modal, Button, Form } from "react-bootstrap";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+const schema = yup.object().shape({
+  name: yup.string().required("Name is required"),
+  subject: yup.string().required("Subject is required"),
+  marks: yup.number().positive().integer().min(0).required("Marks are required"),
+  grades: yup.string().required("Grades are required"),
+});
 
 interface ModalProps {
   visible: boolean;
   setVisible: Function;
+  isEdit: boolean;
 }
 
 // function InputModal() {
 const InputModal: React.FC<ModalProps> = (props) => {
+  const initState = {
+    name: "",
+    subject: "",
+    marks: "",
+    grades: "",
+  };
+  const [initialValues, setInitialValues] = useState(initState);
+  const {
+    register,
+    handleSubmit,
+    reset,
+    // formState,
+    formState: { errors, isSubmitSuccessful },
+  } = useForm({
+    mode: "onTouched",
+    reValidateMode: "onChange",
+    resolver: yupResolver(schema),
+    defaultValues: initialValues,
+  });
+
+  useEffect(() => {
+    if (isSubmitSuccessful) {
+      reset();
+      console.log("Reset form!");
+    }
+  }, [isSubmitSuccessful, reset]);
+
+  const onSubmit = (values: object) => {
+    console.log("Values:::", values);
+    props.setVisible(false);
+    setInitialValues(initState);
```

Comment:
there's no need to set state here as this is doing nothing.
We need to make rendering as less as possible.